src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/LambdaRewriter.cs

```diff
@@ -1553,7 +1554,7 @@ private BoundNode RewriteLambdaConversion(BoundLambda node)
             // NOTE: We require "lambdaScope != null". 
             //       We do not want to introduce a field into an actual user's class (not a synthetic frame).
             var shouldCacheInLoop = lambdaScope != null &&
-                lambdaScope != _analysis.ScopeParent[node.Body] &&
+                lambdaScope != Analysis.GetScopeParent(_analysis.ScopeTree, node.Body).BoundNode &&
```

Comment:
`node.Body` is a nested function body. The parent is the scope where the nested function is declared. I think this must always be non-null since you can't declare a nested function at the top-level.